woocommerce/woocommerce
# [Experimental] Product filters/overlay nav block redesign

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/50273

- ~I've updated the Overlay Navigation block icon to a generic button icon because we're also using this block as a variation for the trigger button and it didn't seem correct to show an icon of "close" when it actually "opens".~ Each overlay navigation open/close will have it's own unique icon.
- I've implemented the logic to remove the overlay navigation block when the overlay setting is set to "Never".
- I've implemented the logic to retain the user settings when they change their mind and set the overlay setting to "Mobile" or "Always".
- I've implemented the logic to add the overlay navigation block back when the overlay setting is set to "mobile" and "always".
- ~I've removed the overlay button styling from the parent Product Filters (Experimental) block since these settings are already there on the Overlay Navigation (Experimental) block.~ I've moved the icon picker to the "Open overlay navigation only".
- Note that we should probably remove the Navigation block (open overlay) from the Product Filters Overlay as it serve no purpose inside there.
- One thing I didn't do in this PR is update the overlay navigation block description depending on the overlay setting. This was mentioned in the original issue however I am not 100% sure what this means. Please clarify if you know. I can either update this PR or I can add it in a separate PR.
- ~I will add tests once we agree this is the direction to take.~ Tests have been added.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

> [!NOTE]
> **Test by developers only**
>
> **Pre-req:**
> Ensure you have the WC Beta Tester Plugin installed and go to Tools > WCA Test Helper > Features. Toggle on the experimental-blocks option.

1. Go to the site editor and navigate to Patterns->General template parts->Product Filters (Experimental). Make sure it has been reseted.
2. Ensure you don't see the Overlay Navigation block in the list view and on the canvas.
3. Select the parent Product Filters (Experimental) and in the inspector controls > Overlay change it to Mobile.
4. Ensure you now see the Overlay Navigation block added in the list view and on the canvas.
5. Test step 3 again but with Always. You should also see the same as step 4.
6. Select the Overlay Navigation (Experimental) block and make some changes to the inspector controls Style.
7. Select the parent Product Filters (Experimental) and in the inspector controls > Overlay change it to Never.
8. Ensure the Overlay Navigation block is now removed.
9. Select the parent Product Filters (Experimental) and in the inspector controls > Overlay change it to Mobile.
10. Ensure you now see the Overlay Navigation block added in the list view and on the canvas and that all the styles you made before in step 6 is retained.


### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Product Filters: update overlay navigation UX
</details>
